src/packet/packet_tls.h

```diff
@@ -215,16 +215,11 @@ struct PreMasterSecret
     guint8 random[46];
 };
 
-struct EncryptedPreMasterSecret
-{
-    guint8 pre_master_secret[128];
-};
-
 //! ClientKeyExchange type in Handshake records
 struct ClientKeyExchange
 {
     uint16 length;
-    struct EncryptedPreMasterSecret exchange_keys;
+    guint8 pre_master_secret[];
```

Comment:
I prefer to keepn these structures names and existance, even if they are not used.
They take the names from RFC4346, that defines:

```
      struct {
          public-key-encrypted PreMasterSecret pre_master_secret;
      } EncryptedPreMasterSecret;
```

Won't be enough to remove the 128 from EncryptedPreMasterSecret?

```
struct EncryptedPreMasterSecret
{
    guint8 pre_master_secret[];
};
```

Reply:
Tried but got flexible array member in a struct with no named members.

Reply:
It's ok then, the other option is to use a pointer and allocate required memory, but that is a bit more complicated and can be done later.